tests/50federation/37auth-events.pl

```diff
@@ -0,0 +1,182 @@
+            )->then( sub {
+               my ( $event ) = @_;
+               log_if_fail( "Received power_levels event id " .
+                  $event->{event_id} );
+               $room->insert_event( $event );
+               Future->done(1);
+            }),
+
+            matrix_put_room_state(
+               $synapse_user,
+               $room_id,
+               type      => "m.room.power_levels",
+               state_key => "",
+               content   => {
+                  ban   => 50,
+                  users => {
+                     $sytest_user_id_a      => 100,
+                     $synapse_user->user_id => 100,
+                  },
+               },
+            ),
+         );
+      })->then( sub {
+         # now let's send an event from the sytest side which only references
+         # the *first* powerlevel event in its auth_events
+
+         my @auth_events = grep { defined } (
+            $room->get_current_state_event( "m.room.create" ),
+            $room->get_current_state_event( "m.room.join_rules" ),
+            $room->get_current_state_event( "m.room.member", $sytest_user_id_a ),
+            $powerlevel_event_a,
+         );
+
+         my $ev = $room->create_event(
+            sender    => $sytest_user_id_a,
+            type      => "m.room.message",
+            content   => {
+               body => "bzzt",
+            },
+            auth_events => SyTest::Federation::Room::make_event_refs( @auth_events ),
+         );
+         $reset_event_id = $ev->{event_id};
+
+         log_if_fail( "Sent message event id " . $reset_event_id );
+
+         $outbound_client->send_event(
+            event => $ev,
+            destination => $synapse_server_name,
+         );
+      })->then( sub {
+         await_sync_timeline_contains( $synapse_user, $room_id, check => sub {
+            my ( $event ) = @_;
+            return unless $event->{type} eq "m.room.message";
+
+            assert_eq( $event->{content}{body}, "bzzt",
+               'event content body' );
+
+            return 1;
+         });
+      })->then( sub {
+         # check the room state has been reset
+         matrix_get_room_state(
+            $synapse_user, $room_id, type => 'm.room.power_levels'
+         )->then( sub {
+            my ( $state ) = @_;
+            log_if_fail( "power_levels in room state after reset", $state );
+            assert_deeply_eq(
+               $state, $powerlevel_event_a->{content}, "room state",
+            );
+            Future->done(0);
+         });
+      })->then( sub {
+         # check that the context has been reset
+         do_request_json_for( $synapse_user,
+            method  => "GET",
+            uri     => "/r0/rooms/$room_id/context/$reset_event_id",
+            params  => { limit => 0 },
+         )->then( sub {
+            my ( $body ) = @_;
+            log_if_fail( "context after reset", $body );
+            Future->done(0);
+         });
```

Comment:
I don't follow what this is testing?

Reply:
oh right. I guess that's why it's been on my hdd rather than PRed for the last 3 months :/